doc/README.md

```diff
@@ -0,0 +1,51 @@
+
+There are three major categories of contracts in this repo:
+
+1. [Tokenlon Protocol Infrastructure](#Protocol-Infrastructure)
+
+2. [Executing Contracts](#Executing-Contracts)
+
+3. [LON Token Peripherals](#LON-Token-Peripherals)
+
+![image info](../tokenlon_architecture.png)
+
+# Protocol Infrastructure
+
+## UserProxy & Tokenlon
+
+User proxy is the entry of the whole procotol while Tokenlon is a transparent upgradeable proxy of it. User proxy navigate users to a specific strategy contract. Meanwhile, it has a multicall entry which allows batching calls between differnt strategy contracts in a single transaction.
+
+## PermanentStorage & ProxyPermanentStorage
+
+In order to manage global states of upgradeable contracts, a permanaent storage is needed in the system. The storage contract itself is also upgradeable considering the flexibility of storing capabilities. The `ProxyPermanentStorage` contract is a transparent upgradeable proxy and `PermanentStorage` is the logic implementation.
+
+## AllowanceTarget & Spender
+
+Several token approvals are required if a user wants to interact with differnt AMM or other trading protocols. It's important to have an unified token authorizing mechanism in Tokenlon since multiple protocols are involved and most of contracts in the Tokenlon are upgradeable as well. Becuase of that, it takes only one token approval from users to enjoy multiple source of liquidity at the same time. The mechanism is made by two parts. The `AllowanceTarget` contract is the target for users to delegate their tokens while `Spender` contract is the only allowed contract that can move funds. `AllowanceTarget` contract has a general `executeCall()` function which can execute arbitrary operations but only `Spender` contract can call it. It supports updating the address of `Spender` contract by the owner as well.
+
+## SpenderSimulation
+
+`SpenderSimulation` is a contract that simulate the result of `Spender` contract. It allows off-chain servies to verify whether traders have properly approved tokens to `AllowanceTarget` contract without actually execute a complete transaction or consume gas for minded transactions. This contract is made for off-chain executing only.
+
+## MarketMakerProxy
+
+MarketMakerProxy is an example implementation of a EIP-1271 compatable contract wallet. It helps market makers or advanced users to onboard Tokenlon more easily.
+
+# Executing Contracts
+
+-   AMMWrapper & AMMQuoter
+-   [RFQ](./executing/RFQ.md)
+-   Limit Order
+-   L2Deposit
```

Comment:
Shall we take L2Deposit apart from Executing Contract, list it under `L2 Related Contracts` or `Miscellaneous Contracts` and finally rename `Executing Contracts` to `Trading Contracts` or `Strategy Contracts`?

Reply:
Then we still need a way to describe those contracts as a whole. `Miscellaneous Contracts` and `Trading Contracts` can be subsets of that.